src/UIState/Wait.cpp

```diff
@@ -14,8 +21,27 @@ Wait::Wait(TankController *tc, uint16_t msDelay, UIState *nextState) : UIState(t
   }
 }
 
+/**
+ * @brief Destroy the Wait:: Wait object
+ *
+ */
+Wait::~Wait() {
+  if (nextState) {
+    delete nextState;
+    nextState = nullptr;
+  }
+}
+
+bool Wait::isInCalibration() {
+  if (nextState) {
```

Comment:
This could be "simplified" (or at least fewer lines):
return nextState ? nextState->isInCalibration() : false;